core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -72,8 +73,14 @@ export class OptimizedSsrEngine {
     this.ssrOptions = ssrOptions
       ? {
           ...defaultSsrOptimizationOptions,
+
           // overrides the default options
           ...ssrOptions,
+          // deep merge the feature toggles
+          featureToggles: deepMerge(
+            { ...defaultSsrOptimizationOptions.featureToggles },
+            ssrOptions.featureToggles
+          ),
```

Comment:
`featutreToggles` object is 1-level shallow. No need for `deepMerge`.
We can either `deepMerge` the whole object with ssr options with default ssr options.
OR
just spread explicitly on the level of `featureToggles`. I don't mind either of the options

```suggestion
          featureToggles: {
            ...defaultSsrOptimizationOptions.featureToggles,
            ...ssrOptions.featureToggles
          },
```